core/insertion_marker_manager.ts

```diff
@@ -608,18 +609,17 @@ export class InsertionMarkerManager {
 
     // Render disconnected from everything else so that we have a valid
     // connection location.
-    insertionMarker.render();
-    insertionMarker.rendered = true;
-    insertionMarker.getSvgRoot().setAttribute('visibility', 'visible');
+    insertionMarker.queueRender();
+    renderManagement.triggerQueuedRenders();
 
-    if (imConn && closest) {
-      // Position so that the existing block doesn't move.
-      insertionMarker.positionNearConnection(imConn, closest);
-    }
-    if (closest) {
-      // Connect() also renders the insertion marker.
-      imConn.connect(closest);
-    }
+    // Position so that the existing block doesn't move.
```

Comment:
I can see why `imConn` has to exist (we throw an error above if it doesn't) but don't see where we're otherwise validating that `closest` exists. Maybe worth keeping the check? or am i missing where we're checking?

Reply:
`closest` is non-nullable! It is always guaranteed to be part of the `activeCandidate`.